paper/paper.md

```diff
@@ -0,0 +1,109 @@
+
+### Heuristic Implementation Example
+
+```kotlin
+override fun simulate(node: NodeType): Double {
+    /*... Original Simulation code ...*/
+    while(true) {
+        val validActions = mdp.actions(currentState)
+        var bestActionScore = Int.MIN_VALUE // Begin heuristic
+            var bestActions = mutableListOf<Point>()
+            for (action in validActions) {
+                val score = heuristicWeight[action.x][action.y]
+                if (score > bestActionScore) {
+                    bestActionScore = score
+                    bestActions = mutableListOf(action)
+                }
+                if (score == bestActionScore) {bestActions.add(action)}
+            }
+        val randomAction = bestActions.random() // End heuristic
+        val newState = mdp.transition(currentState, randomAction)
+        /*... Original Simulation code ...*/
+    }
+}
+```
+
+# Results
+
+When the MCTS solver is accurately selecting the optimal solutions, it will continue to compel the agent to explore in the optimal subset of the state space, and reduce its exploration in the non-optimal subset of the state space. We provide a quick example in the MDP Domain of GridWorld, detailed in [@Liu:2021]. The cumulative number of visits corresponding to the optimal policy is proportionally increasing with respect to the number of MCTS iterations. Whereas for non-optimal solutions, the cumulative visits are significantly less because the solver will avoid visiting the non-optimal state subspace. 
```

Comment:
```suggestion
When the MCTS solver is accurately selecting the optimal solutions, it will continue to compel the agent to explore in the optimal subset of the state space, and reduce its exploration in the non-optimal subset of the state space. We provide a quick example in the MDP Domain of GridWorld, detailed in [@Liu:2021]. The cumulative number of visits corresponding to the optimal policy is proportionally increasing with respect to the number of MCTS iterations. Whereas for non-optimal solutions, the cumulative visits are significantly lower because the solver will avoid visiting the non-optimal state subspace. 
```